OpenRA.Mods.Common/Traits/Attack/AttackDetonates.cs

```diff
@@ -77,11 +70,23 @@ public void ResolveOrder(Actor self, Order order)
 					self.CancelActivity();
 
 				self.SetTargetLine(target, Color.Red);
-				self.QueueActivity(new MoveAdjacentTo(self, target));
-				self.QueueActivity(new CallFunc(() => Explode(self)));
+
+				if (self.Info.HasTraitInfo<MobileInfo>())
+					self.QueueActivity(new MoveAdjacentTo(self, target));
+				else
+				{
+					var aircraft = self.TraitOrDefault<Aircraft>();
+					if (aircraft != null)
```

Comment:
This could be done as:
```c#
				var move = self.TraitOrDefault<IMove>();
				if (move is Mobile)
					self.QueueActivity(new MoveAdjacentTo(self, target));
				else
				{
					var aircraft = move as Aircraft;
```